adr/ADR-32.md

```diff
@@ -197,18 +197,40 @@ Pending messages and bytes count should be updated when:
 - A new pull request is published - add a value of `request.batch_size` to the pending messages count and
 the value of `request.max_bytes` to the pending byte count.
 - A new user message is processed - subtract 1 from pending messages count and subtract message size from penging byte count.
-The message size (in bytes) is defined as: `len(msg.data) + len(msg.subject) + len(msg.reply)`
-- A pull request termination error is received - subtract the value of `Nats-Pending-Messages` header
-from pending messages count and subtract the value of `Nats-Pending-Bytes` from pending bytes count.
-
-###### JetStream error handling
-
-There are 2 kinds of errors which can be received as a response to a pull request:
-
-- Pull request termination error is any error containing `Nats-Pending-Messages` and `Nats-Pending-Bytes` headers.
-These should not be treated as errors by the application, but rather used to calculate the pending messages/bytes count.
-- Every other pull request error should be treated as terminal error - it should be telegraphed to the user (in langie-specific way),
-the `Consume()` execution should be stopped and subscription should be drained.
+- A pull request termination status is received containing `Nats-Pending-Messages` and `Nats-Pending-Bytes` headers, subtract the value of `Nats-Pending-Messages` header from pending messages count and subtract the value of `Nats-Pending-Bytes` from pending bytes count.
+  - 408 Request Timeout
+  - 409 Message Size Exceeds MaxBytes 
+- A pull request termination status of 404 No Messages is received, subtract the entire pull's batch size and max bytes from the pending messages / bytes count.
+
+###### Message Size Calculation
+The message size (in bytes) should be calculated as the server does it. From consumer.go:
+    ```
+    // Calculate payload size. This can be calculated on client side.
+    // We do not include transport subject here since not generally known on client.
+    sz = len(pmsg.subj) + len(ackReply) + len(pmsg.hdr) + len(pmsg.msg)
+    ```
+
+###### Status handling
+
+In addition to possibly providing termination `Nats-Pending-Messages` and `Nats-Pending-Bytes` headers, 
+status messages indicate the termination of the pull.
+Statuses that are errors or warnings should be telegraphed to the user in language specific way.
+
+Errors:
+- 400 Bad Request
+- 409 Consumer Deleted
+- 409 Consumer is push based
+
+Warnings:
```

Comment:
None of these need to be telegraphed to the client (not saying whether we should or not, it is just that they are all recoverable.

- if the pull request all the sudden has a message that fits, it will be yielded
- the max request batch is an error, but if you change the consumer configuration, the client will start working
- the request expires is also controlled by configuration on the consumer that can change out of band
- the max waiting will drop off if some messages are acked, etc.

Reply:
I suppose I could change to say Warnings are not required to be telegraphed